mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -283,11 +294,11 @@ private Mailbox getCachedMailbox(MaildirId mailboxId) throws MailboxNotFoundExce
         if (mailboxId == null) {
             throw new MailboxNotFoundException("null");
         }
-        try {
-            return mailboxCache.get(mailboxId.getRawId());
-        } catch (IndexOutOfBoundsException e) {
-            throw new MailboxNotFoundException(mailboxId);
-        }
+        return mailboxCache.keySet()
```

Comment:
by doing this I dont think we can leverage advantages of the map. How setting hash code of a mailbox as a id of that mailbox and use that id be the key of concurrent map?